### ISSUE TITLE
[Bug] Creation And Rebuilding A Filtered Deck Returns Unexpected Results

### ISSUE THREAD

**Issue by LoganConnor44** [ankidroid/Anki-Android#8876]
###### Reproduction Steps

1. Have a standard deck and mark/answer a few cards with Again `1` and Hard `2` 
    * This may be done on any client
2.  On Anki Desktop, open the Browse tab and create a search string to pull in cards that have been marked/answered again or hard from a specific deck.
    * e.g. `"deck:HSK Vocabulary" AND "rated:7:1" OR "deck:HSK Vocabulary" AND "rated:7:2"`
3.  On Anki Desktop, create a Filtered Deck with step 2's search string
    * Initial creation *and* rebuilding of a filtered deck on AnkiDroid either returns limited results or no results
4.  Sync on Desktop, then proceed to sync on Droid
5. On Anki Droid, locate the filtered deck and notice the card count, then rebuild 
    * Limited or no results will be returned to the deck
6. Sync on Droid, then proceed to sync on Desktop
7. Rebuild the same filtered deck on Anki Desktop
8. Notice the differing results

###### Expected Result
Creation and/or rebuilding a filtered deck should return the same results as Anki Desktop.

###### Actual Result
Creating and/or rebuilding a filtered deck returns limited or no results.

###### Debug info

AnkiDroid Version = 2.14.6
Android Version = 11
ACRA UUID = 949f27f6-ed60-4634-a358-3a5ca7074fd9
Scheduler = std2
Crash Reports Enabled = true

###### Research
*Enter an [x] character to confirm the points below:*

- [x] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [x] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [x] I have searched for similar existing issues here and on the user forum
* The closest issue that may be similar is [this issue](https://github.com/ankidroid/Anki-Android/issues/5308)
- [ ] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

###### Educated Guess
1. My guess would be that the AnkiDroid search string is not evaluated the same as the version on the Desktop.
    * I've played around with the search string a few times, but to no avail, I couldn't find anything that was behaving the same way. Maybe something during an ORM search or a regex difference between python or java.

**Comment by LoganConnor44** [comment]
> Hey @LoganConnor44 - can you please check this against the beta? 

Will do.

**Comment by LoganConnor44** [comment]
Unfortunately, even with beta the issue is still present. 

#Debug Info

AnkiDroid Version = 2.15.0beta4
Android Version = 11
Manufacturer = OnePlus
Model = GM1917
Hardware = qcom
Webview User Agent = Mozilla/5.0 (Linux; Android 11; GM1917 Build/RKQ1.201022.002; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/90.0.4430.210 Mobile Safari/537.36
ACRA UUID = 949f27f6-ed60-4634-a358-3a5ca7074fd9
Scheduler = std2
Crash Reports Enabled = true
DatabaseV2 Enabled = true

**Comment by mikehardy** [comment]
Thank you very much checking! Hopefully the beta is at least nicer in other ways ;-). Tagging it up for a look

**Comment by mikehardy** [comment]
Noting this was also present on 2.14 so this is not a blocker for 2.15 release as it is not a regression although of course we love fixing bugs anytime

**Comment by LoganConnor44** [comment]
For anyone that may view this issue, [this method](https://github.com/ankidroid/Anki-Android/blob/60476b8331c63f42050d0142342b5c410aaeb11d/AnkiDroid/src/main/java/com/ichi2/libanki/Finder.java#L96) looks suspicious to me.

**Comment by mikehardy** [comment]
![image](https://user-images.githubusercontent.com/782704/119074873-6f275580-b9b5-11eb-8cbe-fa61f7a830ac.png)

**Comment by timathon** [comment]
AnkiDroid is getting the "AND" operator wrong.
For example, in Anki desktop, when filtered by
```
"deck:HSK Vocabulary" AND "rated:7:1"
```
In order to get the same result, in AnkiDroid, we need to delete AND from the query, that is 
```
"deck:HSK Vocabulary" "rated:7:1"
```
.
The "OR" operator should be fine.

**Comment by LoganConnor44** [comment]
Will test this weekend.

**Comment by david-allison-1** [comment]
Likely closed in #9733
